src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -210,13 +210,16 @@ public boolean isPinned()
         return getTextChannel().removeReactionById(getId(), unicode, user);
     }
 
-
     @Nonnull
     @Override
     public ReactionPaginationAction retrieveReactionUsers(@Nonnull Emote emote)
     {
         Checks.notNull(emote, "Emote");
 
+        if (getChannelType() == ChannelType.TEXT) {
+            ((TextChannelImpl) channel).checkPermission(Permission.MESSAGE_HISTORY);
```

Comment:
Do we also need to check for the READ permissions? The docs say that we can get a `MISSING_ACCESS` error if we don't have it.

Reply:
You can't have the message history permissions without read, its an implicit permission.

Reply:
I feel like this cast shouldn't be here, this checkPermission method shouldn't be used outside the `TextChannelImpl` class. You should avoid doing that. The `channel.retrieveReactionUsersById` method should be doing this check anyway, I think switching to that instead would be a better solution that doesn't involve using internal helper methods of the TextChannelImpl class like that. This cast makes unit testing very complicated.